geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/iterators/Z3IteratorTest.scala

```diff
@@ -48,35 +50,50 @@ class Z3IteratorTest extends Specification {
           map: java.util.Map[String, String],
           iteratorEnvironment: IteratorEnvironment): Unit = {}
       override def seek(range: Range, collection: java.util.Collection[ByteSequence], b: Boolean): Unit = {
+        staged = key
         key = null
-        staged = null
       }
       override def hasTop: Boolean = staged != null
     }
 
     val iter = new Z3Iterator
     iter.init(srcIter, Z3Iterator.configure(indexValues, 0, None, 25).getOptions, null)
+    iter.seek(new org.apache.accumulo.core.data.Range(new Key(new Text(k)), null), null, inclusive = false)
     iter
   }
 
   "Z3Iterator" should {
 
-    "iterate on points" >> {
-      "keep in bounds values" >> {
-        val test1 = indexValues.sfc.index(-76.0, 38.5, 500)
-        val row = ByteArrays.toBytes(Array[Byte](0, 0), test1)
-        val iter = iterator(row)
-        iter.next()
-        iter.hasTop must beTrue
+    "serialize iterator version" >> {
```

Comment:
Do we need this? 

Also, should we have tests for verifying any backwards compatibility in the serialization format?  (Feel free to push back.)

Reply:
i just wanted to verify it deserialized like i expected and figured it was worth leaving. i can remove it if you want. testing back compatibility would be a little messy, i'd have to copy the old code into the test module i guess? it's not really a supported use case, but i was trying to not break things as much as possible for people who mix-and-match.

Reply:
Sounds great to me.  Leave it as-is and no need to test serialization compatibility (since we assume that folks are matching remote and local code).